handling-forms/submitting-forms.md

```diff
@@ -0,0 +1,428 @@
+
+<!-- prettier-ignore-start -->
+```js
+    return (
+        <main>
+            <h1>Attendance</h1>
+            <StudentList
+                students={studentData}
+                onUpdateStudent={updateStudentData}
+            ></StudentList>
+            <NewStudentForm
+                addStudentCallback={addStudentData}
+            ></NewStudentForm>
+        </main>
+    );
+```
+<!-- prettier-ignore-end -->
+
+Sofia updates the PropTypes of `NewStudentForm` to now anticipate this prop:
+
+<!-- prettier-ignore-start -->
+```js
+import PropTypes from 'prop-types';
+
+const NewStudentForm = (props) => {
+// ...
+
+NewStudentForm.propTypes = {
+    addStudentCallback: PropTypes.func
+};
+```
+<!-- prettier-ignore-end -->
+
+## Handling Form Submissions 
+
+Sofia's next step is to get `NewStudentForm` to add student data when the form submits.
+
+She starts by creating a new function, `onFormSubmit`:
+
+<!-- prettier-ignore-start -->
+```js
+    const onFormSubmit = (event) => {
+        event.preventDefault();
+
+        props.addStudentCallback({
+            nameData: formFields.name,
+            emailData: formFields.email
+        });
+
+        setFormFields({
+            name: '',
+            email: '',
+        });
+    };
+```
+<!-- prettier-ignore-end -->
+
+Sofia's implementation of `onFormSubmit` looks like this:
+
+1. She uses the passed in `event` object and calls `event.preventDefault()`. This prevents the unwanted default behavior of HTML forms.
+1. She invokes the function `props.addStudentCallback()`. This prop was passed in by `App`, with the value `addStudentData`.
+1. She knows that `props.addStudentCallback` receives an object, `newStudent`. She knows that this object should have the keys `nameData` and `emailData`. She passes in an object literal, where the keys are `nameData` and `emailData`, and the values read from the `formFields` state.
+1. She resets the form by updating the members of `formFields` to empty strings.
+
+### !callout-danger
+
+## Prevent the Form's Default Behavior
+
+HTML forms have default behavior: when a form receives a "submit" event, it will make a GET request. This creates an effect where our web app reloads every time we submit a form! Be sure to include `event.preventDefault();` in any form submission event handler.
+
+### !end-callout
+
+## Verify
+
+Sofia's app now handles the new student form submission!
+
+![](../assets/handling-forms_submitting-forms_newstudentform-callback-diagram.png)
+
+Her last step is to verify her work.
+
+In celebration of her effort and work, she inputs her own name...
+
+![](../assets/handling-forms_submitting-forms_sofia-input.png)
+
+and bam! 💥 A new student showed up! The form even got reset, too!
+
+![](../assets/handling-forms_submitting-forms_new-student.png)
+
+We can appreciate how the list of students updated, too. By allowing one component, `App` in this case, manage all of the student data, the other components could focus on displaying it.
```

Comment:
this was a little unclear to me about what exactly we should be appreciating. I'm taking it as how by separating the data storage from the presentation, we got the re-render behavior for free. Is that close to the intent?
```suggestion
We should also take a moment to appreciate how the list of students updated without writing any additional code. By allowing one component, `App` in this case, to manage all of the student data, the other components could focus on displaying it. All Sofia had to do was get the student list in `App` to update, and the existing presentation components did what they do best: display the data!
```